mailbox/elasticsearch/src/main/java/org/apache/james/mailbox/elasticsearch/json/IndexableMessage.java

```diff
@@ -318,10 +319,20 @@ private IndexableMessage(
     public List<MimePart> getAttachments() {
         return attachments;
     }
-    
-    @JsonProperty(JsonMessageConstants.BCC)
-    public EMailers getBcc() {
-        return bcc;
+
+
+    @JsonProperty(JsonMessageConstants.BCC + JsonMessageConstants.EMailer.NAME)
+    public Set<String> getBcName() {
+        return bcc.getEmailers().stream()
+            .map(EMailer::getName)
+            .collect(Guavate.toImmutableSet());
+    }
+
+    @JsonProperty(JsonMessageConstants.BCC + JsonMessageConstants.EMailer.ADDRESS)
+    public Set<String> getBccAddress() {
+        return bcc.getEmailers().stream()
+            .map(EMailer::getAddress)
+            .collect(Guavate.toImmutableSet());
```

Comment:
You may extract a method `getEmailersProperty` taking into parameter and `EMailers` and the function in the map call.